### PR TITLE
Fixes #779 added s2i link

### LINKED ISSUE #779: WIP DO NOT MERGE: Improvement: Add link to OpenShift S2I Build process doc
The Spring Boot CRUD booster mentions S2i process without proposing a link where the end user can discover what is S2I process means

screenshot 2017-11-06 15 57 46

Remark : This remark is general and not specific to this runtime or mission

### PR DESCRIPTION
**Pull request by lavernw1**
@cmoulliard @rhoads-zach - Please review, Added link for S2I process for missions and related procedures.

### CODE REVIEW

File: docs/topics/templates/document-attributes.adoc [hunk 1]
```diff
@@ -173,3 +173,6 @@
 
 // WildFly Swarm repository (mostly for the Contribution Guide)
 :link-repo-wildfly-swarm: https://github.com/wildfly-swarm/wildfly-swarm/
+
+// Link S2I process (primarily for guides mission interaction sections)
+:link-s2i-process: https://docs.openshift.com/container-platform/3.4/architecture/core_concepts/builds_and_image_streams.html#source-build
```
Comment (astrong1) [inline docs/topics/templates/document-attributes.adoc]: Can you update all the links to OpenShift docs to the latest version? These links go to 2.4 which is very out-of-date. You can use "latest".
https://docs.openshift.com/container-platform/latest/architecture/core_concepts/builds_and_image_streams.html#source-build

Comment (lavernw1) [reply docs/topics/templates/document-attributes.adoc]: @astrong1 yes I'll update. Thanks

Comment (lavernw1) [reply docs/topics/templates/document-attributes.adoc]: @astrong1 @rhoads-zach  Did you see the Updates done per your suggestions.

**Comment by astrong1** [comment]
Yes, thank you

File: docs/topics/appendix-s2i-build-process.adoc [hunk 2]
```diff
@@ -1,16 +1,16 @@
 [[s2i-build-process]]
 = The Source-to-Image (S2I) Build Process
-link:https://docs.openshift.com/container-platform/3.4/architecture/core_concepts/builds_and_image_streams.html#source-build[Source-to-Image] (S2I) is a build tool for generating reproducible Docker-formatted container images from online SCM repositories with application sources.
+link:https://docs.openshift.com/container-platform/latest/architecture/core_concepts/builds_and_image_streams.html#source-build[Source-to-Image] (S2I) is a build tool for generating reproducible Docker-formatted container images from online SCM repositories with application sources.
```
Comment (cmoulliard) [inline docs/topics/appendix-s2i-build-process.adoc]: Why do you say "Docker-formatted container" instead of "Docker container image" ?

Comment (cmoulliard) [inline docs/topics/appendix-s2i-build-process.adoc]: As this doc is an appendix, we should mention that 4 build approaches are proposed like 6 different input type - https://www.dropbox.com/s/2j50sv2yvpa1b2x/Screenshot%202017-11-27%2017.08.27.png?dl=0

https://docs.openshift.com/container-platform/3.6/dev_guide/builds/index.html

File: docs/topics/appendix-s2i-build-process.adoc [hunk 3]
```diff
@@ -1,16 +1,16 @@
...
 With S2I builds, you can easily deliver the latest version of your application into production with shorter build times, decreased resource and network usage, improved security, and a number of other advantages.
-For more information, see the link:https://docs.openshift.com/container-platform/3.4/architecture/core_concepts/builds_and_image_streams.html#source-build[Source-to-Image (S2I) Build] chapter of the OpenShift Container Platform documentation.
+For more information, see the link:https://docs.openshift.com/container-platform/latest/architecture/core_concepts/builds_and_image_streams.html#source-build[Source-to-Image (S2I) Build] chapter of the OpenShift Container Platform documentation.
 
 You must provide three elements to the S2I process:
 
 * The application sources hosted in an online SCM repository, such as GitHub.
 
-* The link:https://docs.openshift.com/container-platform/3.4/creating_images/s2i.html#s2i-scripts[S2I scripts].
+* The link:https://docs.openshift.com/container-platform/latest/creating_images/s2i.html[S2I scripts].
```
Comment (cmoulliard) [inline docs/topics/appendix-s2i-build-process.adoc]: to assemble the final Docker image

File: docs/topics/appendix-s2i-build-process.adoc [hunk 4]
```diff
@@ -1,16 +1,16 @@
...
 
 * The Builder image, which serves as the foundation for the assembled image and provides the ecosystem in which your application is running.
```
Comment (cmoulliard) [inline docs/topics/appendix-s2i-build-process.adoc]: such as setting ENV variables to parameterise the bash script responsible to bootstrap the Java runtime when the docker container will be created

File: docs/topics/getting-started-launchpad-update-booster-manual.adoc [hunk 5]
```diff
@@ -1,6 +1,6 @@
 = Updating Manually
 
-This approach allows you to deploy updates directly from a local version of the GitHub repository created by {launcher}. 
+This approach allows you to deploy updates directly from a local version of the GitHub repository created by {launcher}.
```
Comment (cmoulliard) [inline docs/topics/getting-started-launchpad-update-booster-manual.adoc]: using a project created locally on your laptop from a Github repository

File: docs/topics/getting-started-launchpad-update-booster-manual.adoc [hunk 6]
```diff
@@ -48,14 +48,14 @@ NOTE: You can find your token in your {OpenShiftLocal} Web console. For example,
 $ mvn clean fabric8:deploy -Popenshift -DskipTests
 ----
 +
-This command uses the Fabric8 Maven Plugin to launch the S2I process on your {OpenShiftLocal} and to start the pod.
+This command uses the Fabric8 Maven Plugin to launch the link:{link-s2i-process}[S2I process] on your {OpenShiftLocal} and to start the pod.
```
Comment (cmoulliard) [inline docs/topics/getting-started-launchpad-update-booster-manual.adoc]: to create a BuildConfig on Openshift, where the input used is code compiled locally and pushed as binary stream to the platform (see https://docs.openshift.com/container-platform/3.6/dev_guide/builds/build_inputs.html#binary-source) and next to launch the S2I build on Openshift from the content uploaded and using the openjdk s2i image. If the build succeeds and a Docker image is pushed to the docker registry, then the DeploymentConfig is triggered to create and start the corresponding pod.

File: docs/topics/launcher-review-update-booster-manual.adoc [hunk 7]
```diff
@@ -2,7 +2,7 @@
 
 .Next Steps: Update your Booster Manually
 
-You can build and deploy your booster application using the Fabric8 Maven Plugin to launch the S2I process on OpenShift.
+You can build and deploy your booster application using the Fabric8 Maven Plugin to launch the link:{link-s2i-process}[S2I process] on OpenShift.
```
Comment (cmoulliard) [inline docs/topics/launcher-review-update-booster-manual.adoc]: This sentence should be updated according to my previous remark - correct ?

File: docs/topics/appendix-s2i-build-process.adoc (hunk 2 above)
Comment (lavernw1) [reply docs/topics/appendix-s2i-build-process.adoc]: @cmoulliard I think we used  the definition for Source-to-Image (S2I) Build from the architecture documents.  https://docs.openshift.com/container-platform/latest/architecture/core_concepts/builds_and_image_streams.html#source-build .

Comment (astrong1) [reply docs/topics/appendix-s2i-build-process.adoc]: @cmoulliard …to add onto this, OpenShift's "Docker-formatted container image" terminology comes from the _Container Technology Language and Reference Guide_ produced by Product Marketing.

Comment (lavernw1) [reply docs/topics/appendix-s2i-build-process.adoc]: this is per the guidelines:  "container image" or "docker-formatted container ".

File: docs/topics/getting-started-launchpad-update-booster-manual.adoc (hunk 5 above)
Comment (tradej) [reply docs/topics/getting-started-launchpad-update-booster-manual.adoc]: If you accept this change, please replace _laptop_ with _machine_.

File: docs/topics/appendix-s2i-build-process.adoc (hunk 3 above)
Comment (lavernw1) [reply docs/topics/appendix-s2i-build-process.adoc]: @cmoulliard  --  will use" assemble the final container image".

File: docs/topics/appendix-s2i-build-process.adoc (hunk 4 above)
Comment (lavernw1) [reply docs/topics/appendix-s2i-build-process.adoc]: @cmoulliard I'll add a sentence that includes environment variables and parameters used by S2I scripts.  

However, the next sentence in this section (below)  provides a general description of how the builder image is used..
The process injects your application source and dependencies into the Builder image according to instructions specified in the S2I script, and generates a Docker-formatted container image that runs the assembled application.

File: docs/topics/getting-started-launchpad-update-booster-manual.adoc (hunk 5 above)
Comment (lavernw1) [reply docs/topics/getting-started-launchpad-update-booster-manual.adoc]: @cmoulliard not sure how the wording you suggest is different. -- we're are saying this is a local version of the GitHub repository.

File: docs/topics/getting-started-launchpad-update-booster-manual.adoc (hunk 6 above)
Comment (lavernw1) [reply docs/topics/getting-started-launchpad-update-booster-manual.adoc]: @cmoulliard changed  link to reference the s2i process appendix  which summarizes this information.  Did same for the comment in docs/topics/launcher-review-update-booster-manual.adoc.

File: docs/topics/launcher-review-update-booster-manual.adoc (hunk 7 above)
Comment (lavernw1) [reply docs/topics/launcher-review-update-booster-manual.adoc]: see prior comment.

**Comment by lavernw1** [comment]
@cmoulliard did you get a chance to look at the changes.

**Comment by lavernw1** [comment]
@rhoads-zach used link to the S2I appendix in these launcher related  files for references to S2I. Can you review to ensure thats ok.
-  docs/topics/launcher-review-update-booster-manual.adoc
- docs/topics/getting-started-launchpad-update-booster-manual.adoc

**Comment by lavernw1** [comment]
@rhoads-zach removed extra "}"

**Comment by rhoads-zach** [comment]
merged: 872498fece678d6be1292dee76421d9de326b30f

**Pull request closed without merging**